temporalio/converter.py

```diff
@@ -874,6 +874,14 @@ def value_to_type(hint: Type, value: Any) -> Any:
             )
         return hint(value)
 
+    # StrEnum
+    if inspect.isclass(hint) and issubclass(hint, StrEnum):
```

Comment:
@cretz,
Oops, I forgot StrEnum is not in the older Python version.
We can change it to
```python
if inspect.isclass(hint) and issubclass(hint, str) and isinstance(hint, Enum):
```

ref)
![image](https://user-images.githubusercontent.com/3983230/199246770-31a9e932-659f-41b7-a677-29329727fe8d.png)

Reply:
It may be better if we did a:

```python
if sys.version_info >= (3, 11):
  from enum import StrEnum
```

And put that same conditional around this conversion code and around the test

Reply:
Yup that would work too. Let me update my commit

Reply:
Added another commit accordingly. Please kindly review the new commit :pray: